solver/rebalance/rebalance.go

```diff
@@ -38,109 +37,48 @@ func rebalanceForever(
 	backends ethbackend.Backends,
 	solver common.Address,
 ) {
-	for _, chain := range network.EVMChains() {
-		ctx := log.WithCtx(ctx, "chain", evmchain.Name(chain.ID))
-
-		go swapSurplusForever(ctx, interval, backends, solver, chain.ID)
-		go sendSurplusForever(ctx, interval, db, network, backends, solver, chain.ID)
-		go fillDeficitForever(ctx, interval, db, network, pricer, backends, solver, chain.ID)
-	}
-}
-
-// swapSurplusForever swaps surplus tokens to USDC on `chainID` forever.
-func swapSurplusForever(
-	ctx context.Context,
-	interval time.Duration,
-	backends ethbackend.Backends,
-	solver common.Address,
-	chainID uint64,
-) {
-	ctx = log.WithCtx(ctx, "loop", "swapSurplus")
-
-	ticker := time.NewTimer(0)
-	defer ticker.Stop()
-
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-ticker.C:
-			ticker.Reset(interval)
-
-			do := func() error {
-				return swapSurplusOnce(ctx, backends, chainID, solver)
-			}
-
-			if err := expbackoff.Retry(ctx, do); err != nil {
-				log.Error(ctx, "Swap surplus failed", err)
+	go func() {
+		for {
+			start := time.Now()
+			rebalanceOnce(ctx, db, network, pricer, backends, solver)
+			elapsed := time.Since(start)
+
+			// Sleep for the remaining time in the interval, if any.
+			if elapsed < interval {
+				time.Sleep(interval - elapsed)
```

Comment:
rebalance each chain every thirty minutes. if it takes longer than thirty minutes, just do it again right away